azurerm/resource_arm_user_assigned_identity_test.go

```diff
@@ -0,0 +1,45 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAzureRMUserAssignedIdentity_create(t *testing.T) {
+	principalIdRegex := "^[A-Fa-f0-9]{8}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{12}$"
+	resourceName := "azurerm_user_assigned_identity.test"
+	ri := acctest.RandInt()
+	rs := acctest.RandString(14)
+	config := testAccAzureRMUserAssignedIdentityCreate(ri, testLocation(), rs)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineScaleSetDestroy,
```

Comment:
rather than using the VMSS CheckDestroy function - I'm going to push a commit to use one specific for User Assigned Identities (this ensures that when the resource is destroyed by Terraform, it's not lingering around)